lib/services/android-project-service.ts

```diff
@@ -257,13 +258,17 @@ export class AndroidProjectService extends projectServiceBaseLib.PlatformProject
 
 			buildOptions.unshift("buildapk");
 
-			this.$childProcess.on(constants.BUILD_OUTPUT_EVENT_NAME, (data: any) => {
+			const handler = (data: any) => {
 				this.emit(constants.BUILD_OUTPUT_EVENT_NAME, data);
-			});
+			};
 
-			await this.executeGradleCommand(this.getPlatformData(projectData).projectRoot, buildOptions,
-				{ stdio: buildConfig.buildOutputStdio || "inherit" },
-				{ emitOptions: { eventName: constants.BUILD_OUTPUT_EVENT_NAME }, throwError: true });
+			await attachAwaitDetach(constants.BUILD_OUTPUT_EVENT_NAME,
+				this.$childProcess,
+				handler,
+				this.executeGradleCommand(this.getPlatformData(projectData).projectRoot,
```

Comment:
IMO it should be the caller's responsibility to keep track of the arguments. If one of the arguments need to be prepared in some way (e.g. to be `try...catch`-ed) it should be done prior to calling the method.
As for the example you've provided I see no problem with it - failing during evaluation of the arguments will lead to the event handler not being attached in the first place and this method's main target is preventing leakage of event handlers.